### ISSUE TITLE
MagicNumberChecker and custom value types

### ISSUE THREAD

**Issue by athasach** [scalastyle/scalastyle#95]
I have a case where I have a custom Size2D class

```
case class Size2D(width: Double, height: Double)
```

and I would like to be able to declare it as a constant like this

```
class X {
  val FooSize: Size2D = Size2D(640.0, 320.0)
}
```

without getting the warnings for MagicNumberChecker in class X. I can see this kind of magic number case occurring for custom value types, such as a complex number class. Is there some way to suppress this warning for my case? To solve it currently, I would have to pull each constant out of FooSize and declare them separately, but this solution isn't ideal, because FooSize is the constant I want to declare.

Thanks
